### PR TITLE
Active learning bugs

### PR DESCRIPTION
**Pull request by amburger66**
Fixes:
- make sure student only asks about predicates to learn
- for states that have no ground atoms samples, just have an empty set instead of failing
- new dataset type to better ensure that # states = # set of ground atoms in each trajectory

### CODE REVIEW

File: src/approaches/interactive_learning_approach.py [hunk 1]
```diff
@@ -116,29 +108,30 @@ def _relearn_predicates_and_operators(self) -> None:
...
             negative_examples = []
             # Positive examples
-            for i, trajectory in enumerate(self._ground_atom_dataset):
-                for j, ground_atom_set in enumerate(trajectory):
-                    state = self._dataset[i][0][j]
+            for (states, _, ground_atom_sets) in self._dataset:
+                assert len(states) == len(ground_atom_sets)
+                for (state, ground_atom_set) in zip(states, ground_atom_sets):
+                    if len(ground_atom_set) == 0:
+                        continue  # pragma: no cover
```
Comment (ronuchit) [inline src/approaches/interactive_learning_approach.py:115]: Is there a reason we can't test this? E.g. if we give data that contains states but we have no predicates, then ground_atom_set should always be empty, and we can test that there are no positives?

**Review by ronuchit — COMMENTED** [review]
lgtm

**Comment by tomsilver** [comment]
LGTM too!

File: src/approaches/interactive_learning_approach.py (hunk 1 above)
Comment (amburger66) [reply src/approaches/interactive_learning_approach.py:115]: good point! how is this?

Comment (ronuchit) [reply src/approaches/interactive_learning_approach.py:115]: if the test is working correctly, we should be able to remove this "pragma: no cover" right?

Comment (amburger66) [reply src/approaches/interactive_learning_approach.py]: right, i forgot to remove it

**Pull request merged**